pkg/build/strategies/sti/sti.go

```diff
@@ -287,7 +284,7 @@ func (b *STI) Exists(config *api.Config) bool {
 		}
 	}
 
-	return previousImageExists && b.installedScripts[api.SaveArtifacts]
+	return previousImageExists
```

Comment:
Not always true ;) It'll be always true for images with embedded scripts, because in that case the only option to verify save-artifacts existence is actually running that script. In all other cases it depends on whether that script was downloaded from one of the possible locations (external URL, sources).